### PR TITLE
Introduce memory compaction

### PR DESCRIPTION
**Pull request by antisaling**

### CODE REVIEW

File: commands/simple/Makefile [hunk 1]
```diff
@@ -536,6 +537,9 @@ mesg:	mesg.c
 	$(CCLD) -o $@ $?
 	@install -S 4kw $@
 
+memstat:	memstat.c
+	$(CCLD) -o $@ $? -lcurses
```
Comment (antisaling) [inline commands/simple/Makefile]: does it still use curses?

File: commands/simple/memstat.c [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+			h_len);
+	}
+}
+
+int main(int argc, char *argv[])
+{
+	int i, p;
+	struct proc proc[PROCS];
+        struct pm_mem_info pmi;
+	int prev_uptime, uptime;
```
Comment (antisaling) [inline commands/simple/memstat.c]: no longer need these

Comment (isinyaaa) [reply commands/simple/memstat.c:175]: ditto

File: commands/simple/memstat.c [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+		printf(" === [ hole ] === %8ldK %8ldK %8ldK\n",
+			h_base,
+			h_base + h_len,
+			h_len);
+	}
+}
+
+int main(int argc, char *argv[])
+{
+	int i, p;
```
Comment (antisaling) [inline commands/simple/memstat.c]: ditto

File: servers/pm/alloc.c [hunk 4]
```diff
@@ -28,13 +28,15 @@
 #include "../../kernel/const.h"
 #include "../../kernel/config.h"
 #include "../../kernel/type.h"
+#define ENABLE_SWAP 1
```
Comment (antisaling) [inline servers/pm/alloc.c]: todo: comment why this is here

**Pull request closed without merging**